### PR TITLE
Fix report statistics render differences

### LINKED ISSUE #3150: Calendar of Report Statistics renderers and popups
When rendering reports in a report statistics calendar vs the calendar section of Reports, items are displayed differently. Furthermore, users can't expand the events displayed in the report statistics calendar.

Make sure that when clicking on events in the report statistics calendar, items are opened.

### PR DESCRIPTION
**Pull request by cemalettin-simsoft**
Default calendar widget was accepting only 'count per date'. Now it accepts array of objects(array of reports). This fixes popover issue and when you click on an event, it goes to that event's url.

report-statistics

### Release notes

Closes #3150 

#### User changes
- Report statistics and report calendar become more consistent

#### System admin changes
- [ ] anet.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [X] Described the user behavior in PR body
  - [X] Referenced/updated all related issues
  - [X] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [X] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [ ] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

**Review by VassilIordanov — APPROVED** [review]

File: client/src/components/aggregations/CalendarWidget.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {
   aggregationWidgetDefaultProps,
   aggregationWidgetPropTypes
 } from "components/aggregations/utils"
+import { reportsToEvents } from "components/ReportCalendar"
```
Comment (gjvoosten) [inline client/src/components/aggregations/CalendarWidget.js:9]: Hm, I don't like the fact that this widget, which is itself not report-specific, imports from `ReportCalendar`. I suggest to refactor this.

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/src/components/aggregations/CalendarWidget.js (hunk 1 above)
Comment (cemalettin-simsoft) [reply client/src/components/aggregations/CalendarWidget.js:9]: Yes, i was planning to ask this part, we need type of objects information + how to make them events, something like this:
`GET_EVENTS_FUNCTIONS = {
  Report: reportsToEvents,
  Task: tasksToEvents, // will be implemented when needed
 etc...
}
`

i will try to implement this, maybe create another file where all the object => calendarEvent transformations exist

Comment (gjvoosten) [reply client/src/components/aggregations/CalendarWidget.js:9]: These transformations might go into the already existing `client/src/components/aggregations/utils.js` (seems as good a place as any).

File: client/src/components/aggregations/utils.js [hunk 2]
```diff
@@ -188,3 +189,34 @@ export const likertScaleAndPieAggregation = (fieldName, fieldConfig, data) => {
     }
   }
 }
+
+export const CALENDAR_OBJECT_TYPES = {
+  Report: "Report"
```
Comment (gjvoosten) [inline client/src/components/aggregations/utils.js:194]: In keeping with the convention in other places, I prefer:
```
  REPORT: "Report"
```

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/src/components/aggregations/utils.js (hunk 2 above)
Comment (cemalettin-simsoft) [reply client/src/components/aggregations/utils.js:194]: Done

File: client/src/components/aggregations/utils.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import _clone from "lodash/clone"
 import _isEmpty from "lodash/isEmpty"
+import { Person, Report } from "models"
 import moment from "moment"
 import { AssessmentPeriodPropType, PeriodPropType } from "periodUtils"
 import PropTypes from "prop-types"
 import Settings from "settings"
-
```
Comment (gjvoosten) [inline client/src/components/aggregations/utils.js:7]: Please keep a blank line after the imports.

File: client/src/components/aggregations/CalendarWidget.js [hunk 4]
```diff
@@ -63,6 +59,11 @@ const CalendarWidget = ({
       aspectRatio={3} // ratio of width-to-height
       ref={calendarComponentRef}
       events={events}
+      eventClick={info => {
+        history.pushState(info.event.url)
```
Comment (gjvoosten) [inline client/src/components/aggregations/CalendarWidget.js:63]: Just `history.push(info.event.url)` please.

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/src/components/aggregations/CalendarWidget.js (hunk 4 above)
Comment (cemalettin-simsoft) [reply client/src/components/aggregations/CalendarWidget.js:63]: Whoops, autocomplete not always a friend

File: client/src/components/ReportCalendar.js [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
 import Calendar from "components/Calendar"
 import { PageDispatchersPropType } from "components/Page"
 import _isEqual from "lodash/isEqual"
-import { Person, Report } from "models"
 import moment from "moment"
 import PropTypes from "prop-types"
 import React, { useRef } from "react"
 import { useHistory } from "react-router-dom"
-import Settings from "settings"
-
```
Comment (gjvoosten) [inline client/src/components/ReportCalendar.js:12]: Please keep a blank line after the imports.

**Review by gjvoosten — CHANGES_REQUESTED** [review]

**Review by gjvoosten — APPROVED** [review]

**Pull request merged**